sigmf/sigmffile.py

```diff
@@ -405,7 +405,7 @@ def get_annotations(self, index=None):
         return [
             x for x in self._metadata.get(self.ANNOTATION_KEY, [])
             if index is None or (x[self.START_INDEX_KEY] <= index
-            and x[self.START_INDEX_KEY] + x[self.LENGTH_INDEX_KEY] > index)
+            and (self.LENGTH_INDEX_KEY not in x or x[self.START_INDEX_KEY] + x[self.LENGTH_INDEX_KEY] > index))
         ]
```

Comment:
I think 5 lines of logic within a list comprehension is too many. Please break this out so it is more clear like
```python
annots = []
for annot in self._metadata.get(self.ANNOTATION_KEY, []):
   annotation_logic()
return annots
```